src/scripts/main.js

```diff
@@ -1 +1,81 @@
 'use strict';
+
+let leftClicked = false;
+let rightClicked = false;
```

Comment:
Don't use global boolean flags
create a promise per each event 
and combine them with Promise.all